node/pkg/vaa/structs_test.go

```diff
@@ -491,8 +492,161 @@ func TestVerifySignaturesFuzz(t *testing.T) {
+			payloadType:    1,
+			emitterChainId: ChainIDEthereum,
+			emitterAddr:    "0000000000000000000000000290FB167208Af455bB137780163b7B7a9a10C16",
+			tokenChainId:   ChainIDEthereum,
+			tokenAddr:      "000000000000000000000000DDb64fE46a91D46ee29420539FC25FD07c5FEa3E",
+			toChainId:      ChainIDSolana,
+			toAddr:         "21c175fcd8e3a19fe2e0deae96534f0f4e6a896f4df0e3ec5345fe27ac3f63f0",
+			amount:         725000000,
+			errString:      "",
+		},
+		{label: "unsupported payload type",
+			vaa:       "01000000000100e424aef95296cb0f2185f351086c7c0b9cd031d1288f0537d04ab20d5fc709416224b2bd9a8010a81988aa9cb38b378eb915f88b67e32a765928d948dc02077e00000102584a8d000000020000000000000000000000000290fb167208af455bb137780163b7b7a9a10c16000000000000000f0f02000000000000000000000000000000000000000000000000000000002b369f40000000000000000000000000ddb64fe46a91d46ee29420539fc25fd07c5fea3e000221c175fcd8e3a19fe2e0deae96534f0f4e6a896f4df0e3ec5345fe27ac3f63f000010000000000000000000000000000000000000000000000000000000000000000",
+			errString: "unsupported payload type",
+		},
+		{label: "buffer too short",
+			vaa:       "01000000000100e424aef95296cb0f2185f351086c7c0b9cd031d1288f0537d04ab20d5fc709416224b2bd9a8010a81988aa9cb38b378eb915f88b67e32a765928d948dc02077e00000102584a8d000000020000000000000000000000000290fb167208af455bb137780163b7b7a9a10c16000000000000000f0f01",
+			errString: "buffer too short",
+		},
+		{label: "empty string",
+			vaa:       "",
+			errString: "VAA is too short",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(string(tc.label), func(t *testing.T) {
+			data, err := hex.DecodeString(tc.vaa)
+			assert.Nil(t, err)
+
+			vaa, err := Unmarshal(data)
+			if err != nil {
+				assert.Equal(t, tc.errString, err.Error())
+			} else {
+				assert.Nil(t, err)
+				assert.NotNil(t, vaa)
+
+				if len(tc.errString) == 0 {
+					expectedEmitterAddr, err := StringToAddress(tc.emitterAddr)
+					assert.Nil(t, err)
+
+					expectedTokenAddr, err := StringToAddress(tc.tokenAddr)
+					assert.Nil(t, err)
+
+					expectedToAddr, err := StringToAddress(tc.toAddr)
+					assert.Nil(t, err)
+
+					expectedAmount := big.NewInt(tc.amount)
+
+					assert.Equal(t, tc.emitterChainId, vaa.EmitterChain)
+					assert.Equal(t, expectedEmitterAddr, vaa.EmitterAddress)
+					assert.Equal(t, 133, len(vaa.Payload))
+
+					payload, err := DecodeTransferPayloadHdr(vaa.Payload)
+					assert.Nil(t, err)
+					assert.Equal(t, tc.payloadType, payload.Type)
+					assert.Equal(t, tc.tokenChainId, payload.TokenChainID)
+					assert.Equal(t, expectedTokenAddr, payload.TokenAddress)
+					assert.Equal(t, tc.toChainId, payload.ToChainID)
+					assert.Equal(t, expectedToAddr, payload.ToAddress)
+					assert.Equal(t, expectedAmount.Cmp(payload.Amount), 0)
```

Comment:
I almost wonder if this wouldn't make more sense to just have something more like:

```
assert.Equal(t.payload, tc, payload)
```

You'd need to change the Test struct to be dramatically simplified then.